tsan.mixin

```diff
@@ -0,0 +1,10 @@
+{
+    "build": {
+        "tsan-gcc": {
+            "cmake-args": [
+                "-DCMAKE_C_FLAGS='-fsanitize=thread -O1 -g -fno-omit-frame-pointer'",
```

Comment:
Can you please comment why this should pass `-O1`?

Reply:
Here's the reference from https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual I used: 
```
Simply compile your program with -fsanitize=thread and link it with -fsanitize=thread. To get a reasonable performance add -O2. Use -g to get file names and line numbers in the warning messages.
```

Thread sanitizer slows down the runtime of the code. -O1 helped speed this up a bit as per my tests. Updating to -O2 as per the guidance on the above wiki

Reply:
Using the recommended optimization level makes sense to me. If it interferes with our regular tests we can revisit it.